src/internal/pfsload/worker.go

```diff
@@ -0,0 +1,79 @@
+package pfsload
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"github.com/pachyderm/pachyderm/v2/src/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/backoff"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/miscutil"
+	"github.com/pachyderm/pachyderm/v2/src/internal/task"
+	log "github.com/sirupsen/logrus"
+)
+
+const namespace = "pfsload"
+
+func Worker(pachClient *client.APIClient, taskService task.Service) error {
```

Comment:
I think you could get by with just the pfs.APIClient interface instead of the whole client.APIClient.

Reply:
I need the full client for the auth functionality, but with some improvements to auth (primarily refactoring the functionality into utilities) we should be able to do that eventually.